internal/usde/usde.go

```diff
@@ -47,107 +49,163 @@ func ResourceNeedsUpdating(ctx context.Context, newDef, existingDef *unstructure
-	for _, dataLossField := range dataLossFields {
+	switch upgradeStrategy {
+	case apiv1.UpgradeStrategyProgressive:
+		mergedSpecFieldLists := []config.SpecField{}
+		mergedSpecFieldLists = append(mergedSpecFieldLists, dataLossFields...)
+		mergedSpecFieldLists = append(mergedSpecFieldLists, progressiveFields...)
+		mergedSpecFieldLists = append(mergedSpecFieldLists, recreateFields...)
+
+		specNeedsUpdating, err := checkFieldsList(ctx, mergedSpecFieldLists, newDef, existingDef)
+		if err != nil {
+			return false, apiv1.UpgradeStrategyError, false, fmt.Errorf("error while checking spec changes using full USDE Config (strategy '%s'): %w", upgradeStrategy, err)
+		}
+		if specNeedsUpdating {
+			return specNeedsUpdating, upgradeStrategy, false, nil
+		}
+	case apiv1.UpgradeStrategyPPND:
+		// Use the recreate fields list from config
+		specNeedsUpdating, err := checkFieldsList(ctx, recreateFields, newDef, existingDef)
+		if err != nil {
+			return false, apiv1.UpgradeStrategyError, false, fmt.Errorf("error while checking spec changes using 'recreate' USDE Config (strategy '%s'): %w", upgradeStrategy, err)
+		}
+		if specNeedsUpdating {
+			// Also return "recreate" true to communicate to the controller to recreate the appropriate resources
+			return specNeedsUpdating, upgradeStrategy, true, nil
+		}
+
+		// Use the dataLoss fields list from config
+		specNeedsUpdating, err = checkFieldsList(ctx, dataLossFields, newDef, existingDef)
+		if err != nil {
+			return false, apiv1.UpgradeStrategyError, false, fmt.Errorf("error while checking spec changes using 'dataLoss' USDE Config (strategy '%s'): %w", upgradeStrategy, err)
+		}
+		if specNeedsUpdating {
+			return specNeedsUpdating, upgradeStrategy, false, nil
+		}
+
+		// Use the progressive fields list from config and return direct-apply strategy if the spec needs to update
+		specNeedsUpdating, err = checkFieldsList(ctx, progressiveFields, newDef, existingDef)
+		if err != nil {
+			return false, apiv1.UpgradeStrategyError, false, fmt.Errorf("error while checking spec changes using 'progressive' USDE Config (strategy '%s'): %w", upgradeStrategy, err)
+		}
+		if specNeedsUpdating {
+			return specNeedsUpdating, apiv1.UpgradeStrategyApply, false, nil
+		}
+	case apiv1.UpgradeStrategyApply:
+		specNeedsUpdating, err := checkFieldsList(ctx, recreateFields, newDef, existingDef)
+		if err != nil {
+			return false, apiv1.UpgradeStrategyError, false, fmt.Errorf("error while checking spec changes using 'recreate' USDE Config (strategy '%s'): %w", upgradeStrategy, err)
+		}
+		if specNeedsUpdating {
+			// Also return "recreate" true to communicate to the controller to recreate the appropriate resources
+			return specNeedsUpdating, upgradeStrategy, true, nil
+		}
+	}
+
+	numaLogger.Debug("no USDE Config field changes detected, comparing specs for any DirectApply-type of changes")
+
+	// If there were no changes in the data loss fields, there could be changes in other fields of the specs.
```

Comment:
nit: update comment to account for the other types of fields, not just "data loss"